neighsyncd/neighsync.cpp

```diff
@@ -38,8 +41,13 @@ void NeighSync::onMsg(int nlmsg_type, struct nl_object *obj)
 
     key+= LinkCache::getInstance().ifindexToName(rtnl_neigh_get_ifindex(neigh));
     key+= ":";
-    nl_addr2str(rtnl_neigh_get_dst(neigh), addrStr, MAX_ADDR_SIZE);
-    key+= addrStr;
+
+    nl_addr2str(rtnl_neigh_get_dst(neigh), ipStr, MAX_ADDR_SIZE);
+    string ip(ipStr);
+    /* Ignore IPv6 multicast addresses as neighbors */
+    if (family == IPV6_NAME && ip.compare(0, 4, IPV6_LINKLOCAL_SCOPE_MULTICAST_ADDRESSES_PREFIX))
```

Comment:
instead of defining your own macro, I'd like to use existing macro. 

1. include netinet/in.h
2. use nl_addr_fill_sockaddr to convert the a sockaddr,
3. use existing macro IN6_IS_ADDR_MC_LINKLOCAL in in.h to do the check.